src/ExpressReceiver.ts

```diff
@@ -64,25 +68,52 @@ export default class ExpressReceiver implements Receiver {
     // tslint:disable-next-line: align
     }, 3001);
 
-    const event: ReceiverEvent = {
-      body: req.body,
-      ack: async (response: any): Promise<void> => {
-        if (isAcknowledged) {
-          throw new ReceiverMultipleAckError();
-        }
-        isAcknowledged = true;
-        if (!response) {
-          res.send('');
-        } else if (typeof response === 'string') {
-          res.send(response);
-        } else {
-          res.json(response);
-        }
-      },
-    };
+    let event: ReceiverEvent;
+    let storedResponse = undefined;
+    if (this.processBeforeResponse) {
+      event = {
+        body: req.body,
+        ack: async (response): Promise<void> => {
+          if (isAcknowledged) {
+            throw new ReceiverMultipleAckError();
+          }
+          isAcknowledged = true;
+          if (!response) {
+            res.send('');
```

Comment:
The HTTP response should not be sent at this timing. Probably, setting `''` as `storedResponse` instead would be expected.